infrastructure/lib/infrastructure-stack.ts

```diff
@@ -0,0 +1,21 @@
+import * as cdk from "@aws-cdk/core";
+import * as ecs from "@aws-cdk/aws-ecs";
+import * as ecsPatterns from "@aws-cdk/aws-ecs-patterns";
+
+export class InfrastructureStack extends cdk.Stack {
+  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
+    super(scope, id, props);
+
+    new ecsPatterns.ApplicationLoadBalancedFargateService(this, `${id}-ecs`, {
+      ...props,
+      memoryLimitMiB: 1024,
+      cpu: 512,
+      taskImageOptions: {
+        image: ecs.ContainerImage.fromAsset("../", {
+          exclude: ["infrastructure/*"],
+        }),
+        containerPort: 8080,
```

Comment:
:mag: So what's happening here?

Reply:
More info [here](https://docs.aws.amazon.com/AmazonECS/latest/developerguide/task_definition_parameters.html)

`memoryLimitMiB`: pretty obvious; we're limiting to 1 GiB 

`cpu`: the number of CPU units we want to use. It's not clear to me _exactly_ what a "CPU unit" is, but here's a snippet from the link that helps give a sense of the scale of one:
> You can determine the number of CPU units that are available per Amazon EC2 instance type by multiplying the number of vCPUs listed for that instance type on the Amazon EC2 Instances detail page by 1,024.
So, it appears as though 512 is "half a CPU"

`taskImageOptions`: all things docker image. Here we're telling CDK to build the image from an asset (our root dockerfile). We could also tell it to grab one from a public or private repo like docker hub. The port is the port to which incoming traffic is forwarded (i.e. the one the container is listening on)